### PR TITLE
Various modifications of Mesh_3

### PR DESCRIPTION
**Pull request by lrineau**

### CODE REVIEW

File: Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h [hunk 1]
```diff
@@ -406,6 +409,14 @@ detect_features(FT angle_in_degree, std::vector<Polyhedron>& poly)
   if (borders_detected_)
     return;//prevent from not-terminating
 
+  typedef Featured_edges_copy_graph G_copy;
+  G_copy g_copy;
+  typedef typename boost::graph_traits<G_copy>::vertex_descriptor vertex_descriptor;
+  typedef std::map<typename Polyhedron::Point,
+                   vertex_descriptor> P2vmap;
+  // TODO: replace this map by and unordered_map
```
Comment (sloriot) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h]: Did you forget it?

Comment (lrineau) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h]: Not really... Andreas told me that there was no has function for `CGAL::Point_3<K>`. There is an open issue on the subject: https://github.com/CGAL/cgal/issues/1116.

Comment (sloriot) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h]: What are you using it for? detecting duplicated points?

Comment (lrineau) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h]: Yes, exactly: polyhedron vertices (from different polyhedra or not) with the same coordinates are assigned to the same vertex of the graph.

... I could transform our CGAL `Point_3` to `CGAL::cpp11::array<double, 3>` and use the standard hash functions from Boost.

**Comment by lrineau** [comment]
https://github.com/CGAL/cgal/pull/1845/commits/9b05a3a5a90139fa8b5a05fb303e8308b5bdb1cc was not tested.

**Comment by sloriot** [comment]
New warnings, at least [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-Ic-111/Generator/TestReport_cgaltester_x86-64_Darwin-13.0_Apple-clang-5.0_Release.gz)

**Pull request merged**